sass/core/_strings.scss

```
@@ -71,8 +71,16 @@
   @if $string == $old {
     $return: $new;
   } @else if $i {
-    // Force $new & $string to string
-    $new: if((type-of($new) == 'string') or (not $new), $new, '#{$new}');
+    // Convert $new to a string
+    @if ($new and (type-of($new) != 'string')) {
+      @if (type-of($new) == 'map') {
+        @warn '_a_str-replace $new parameter cannot be a map';
+        @return $string;
+      }
+      @else {
```

```suggestion
      } @else {
```

just for formatting consistency…